akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/TimerScheduler.scala

```diff
@@ -165,4 +169,29 @@ trait TimerScheduler[T] {
    */
   def cancelAll(): Unit
 
+  final def asScala: scaladsl.TimerScheduler[T] = this match {
+    case scalaThis: scaladsl.TimerScheduler[T @unchecked] => scalaThis
+    case jSelf =>
+      new scaladsl.TimerScheduler[T] {
+        import akka.util.JavaDurationConverters._
+        override def startTimerWithFixedDelay(key: Any, msg: T, delay: FiniteDuration): Unit =
+          jSelf.startTimerWithFixedDelay(key, msg, delay.asJava)
+
+        override def startTimerAtFixedRate(key: Any, msg: T, interval: FiniteDuration): Unit =
+          jSelf.startTimerAtFixedRate(key, msg, interval.asJava)
+
+        override def startPeriodicTimer(key: Any, msg: T, interval: FiniteDuration): Unit =
+          startTimerWithFixedDelay(key, msg, interval) //this follows the deprecation recommendation
+
+        override def startSingleTimer(key: Any, msg: T, delay: FiniteDuration): Unit =
+          jSelf.startSingleTimer(key, msg, delay.asJava)
+
+        override def isTimerActive(key: Any): Boolean = jSelf.isTimerActive(key)
+
+        override def cancel(key: Any): Unit = jSelf.cancel(key)
+
+        override def cancelAll(): Unit = jSelf.cancelAll()
+      }
+  }
```

Comment:
Can we make these be completely abstract and have the impl implement both and return `this` rather than delegating/creating instances? I think we do that pattern quite a log in the typed internals. See ActorContext/TypedActorContext etc. for one example.

Reply:
is it acceptable to have the one DSL implement the other with `final` methods? this way concrete classes will only have to implement one (say, scala) DSL

Reply:
Better keep the API trait with no logic and have the impl cover both IMO. But if you want to avoid duplication in the testkit you could introduce an abstract impl doing the delegation from Java to Scala in the internal package.